### ISSUE TITLE
Skeleton bone scaling not properly applied on import

### ISSUE THREAD

**Issue by poisenbery** [BadDogSkyrim/PyNifly#76]
If a character has a custom skeleton with scaling applied to certain bones, the scaling is applied to anything that is a parent of that bone, and the other bones are shifted.

IE: If you were to scale the spine0 bone down 20%, the entire upper body mesh would shrink in size and the parented bones will get closer together.

The way it currently works, only the new bone positions are imported into blender; scaling does not properly apply in blender.

If you were to scale the ubody and lbody bones up 10x, it would result in a character that is 10x the normal size. However, if you import this skeleton, only the bone positions get updated, and you end up with a misfigured body.

If scaling can somehow be applied to the bones in pose mode it would solve the issue.

**Comment by poisenbery** [comment]
I was going through the source code and I THINK this is the part that needs adjusting? I'm not a programmer though so my only understanding of this is your annotations.

`# use the transform in the file if there is one; otherwise get the 
        # transform from the reference skeleton
        bone_xform = self.nif.get_node_xform_to_global(self.nif.nif_name(name)) 

        bone = armdata.edit_bones.new(name)
        bone.head = bone_xform.translation
        if self.nif.game in ("SKYRIM", "SKYRIMSE"):
            rot_vec = bone_xform.rotation.by_vector((0.0, 0.0, 5.0))
        else:
            rot_vec = bone_xform.rotation.by_vector((5.0, 0.0, 0.0))
        bone.tail = (bone.head[0] + rot_vec[0], bone.head[1] + rot_vec[1], bone.head[2] + rot_vec[2])
        bone['pyxform'] = bone_xform.rotation.matrix # stash for later

        return bone`
        
        
        It looks like it gets the transform data but not the scale data.

**Comment by BadDogSkyrim** [comment]
True. Can you give me a sample nif where there is scaling on bones? I'm curious as to how that's even useful, given that the bones generally have to match the skeletons.

**Comment by poisenbery** [comment]
I have a lot of CME -> NPC Follower conversions in my collection that I've made over the years. The skeleton_female.nif that I have for the NPC's have scaling on a lot of the bones to do things like adjusting proportions and size of the limbs and body. 



My process for importing is as follows:
1.) Import an unaltered CME Skeleton
2.) Import all body parts and clothing onto the base skeleton
3.) Use NiDumpScale (3rd party program) on the NPC skeleton to see which bones have a scale value that does not equal 0
4.) Scale the bones in blender 1 by 1 in pose mode
5.) Apply visual geometry to mesh and apply pose to rest pose, reparent with empty groups
6.) Create a new object and import the NPC Skeleton [if I import with nothing selected, it seems to combine with the other skeleton]
7.) Use a script to automatically match bone positions from the skeleton I first imported onto the NPC skeleton I imported last.
8.) Delete the skeleton I have imported last [it was only needed to match position]. 
9.) Apply geometry to all objects, apply pose as rest pose to skeleton, reparent with empty groups. 
10.) [Probably my fault in the character creation process] Move the eyes, brows, and mouth to the correct location. [They always appear on the ground with the head in the correct place, not sure why.]

This is the only workflow I have found that allows me to get the exact shape of the NPC as it appears in-game. If I import the body parts with the NPC skeleton instead of an unaltered one, the body parts get positioned in the correct location, but the scaling vaules that you see in nifscope don't seem to appear at all in blender, so it ends up looking very strange. 

It's a very specific thing; AFAIK, nobody else does this.
[Cleo.zip](https://github.com/BadDogSkyrim/PyNifly/files/7583413/Cleo.zip)
 
 Here's a skeleton and output from NiDumpScale that shows which bones were scaled

**Comment by Kalessin42** [comment]
This can matter with custom bones (e.g. for SMP). For instance, https://www.nexusmods.com/skyrimspecialedition/mods/90194 has bones on the earrings, and they're quite small to fit the earrings. Importing the nif for the earrings into blender results in the bones being the wrong size, and they overlap.

So, as things stand, if I don't want bones to be messed up on any SMP stuff that I'm doing, I can't export any such meshes and then reimport them if I don't want to have to fix the bones every time that I reimport the mesh.
